src/bptl/work_units/valid_sign/tasks.py

```diff
@@ -0,0 +1,382 @@
+            method="GET",
+        )
+
+        roles = response["results"]
+        # Not all the roles are signers (one of them is the account owner)
+        return [role for role in roles if role["type"] == "SIGNER"]
+
+    def _get_approvals(self, signers: List[dict]) -> List[dict]:
+        """Make approvals from signers
+
+        The approval is a placeholder for where a signature from a signer will go.
+        """
+
+        fields = [
+            {
+                "type": "SIGNATURE",
+                "subtype": "FULLNAME",
+                "extractAnchor": {
+                    "anchorText": "Capture Signature",
+                    "index": 0,
+                    "characterIndex": 0,
+                    "anchorPoint": "BOTTOMLEFT",
+                    "leftOffset": 0,
+                    "topOffset": 0,
+                    "width": 150,
+                    "height": 50,
+                },
+            }
+        ]
+
+        approvals = []
+        for signer in signers:
+            approvals.append({"role": f"{signer['id']}", "fields": fields})
+
+        return approvals
+
+    def create_package(self) -> dict:
+        """Create a ValidSign package with the name specified by the process variable and add the signers to it."""
+
+        logger.debug("Creating ValidSign package")
+
+        variables = self.task.get_variables()
+        signers = self.format_signers(check_variable(variables, "signers"))
+        package_name = check_variable(variables, "packageName")
+
+        body = {
+            "name": package_name,
+            "type": "PACKAGE",
+            "roles": signers,
+        }
+
+        package = self.get_validsign_client().request(
+            path="api/packages", operation="api.packages.post", method="POST", json=body
+        )
+
+        return package
+
+    def add_documents_and_approvals_to_package(self, package: dict) -> List[dict]:
+        """Add documents and approvals to the package."""
+
+        logger.debug(
+            "Adding documents and approvals to ValidSign package '%s'", package["id"]
+        )
+
+        documents = self._get_documents_from_api()
+
+        # Multiple files can be added in one request by passing the following 'files' parameter
+        # to the request, but then not sure how to specify the filename yet...
+        # files = [("files[]", content) for name, content in documents]
+
+        validsign_client = self.get_validsign_client()
+
+        signers = self._get_signers_from_package(package)
+        approvals = self._get_approvals(signers)
+
+        attached_documents = []
+        for doc_name, doc_content in documents:
+            url = f"{validsign_client.base_url}api/packages/{package['id']}/documents"
+            payload = {"name": doc_name, "extract": True, "approvals": approvals}
+            body = {"payload": json.dumps(payload)}
+            if isinstance(doc_content, TemporaryUploadedFile):
```

Comment:
if you wrap the document data in `BytesIO` above, you don't need this `isinstance` check